rdr_service/model/bq_questionnaires.py

```diff
@@ -76,45 +77,75 @@ def get_fields(self):
 
         dao = BigQuerySyncDao(backup=True)
 
-        # Load module field data from the code table if available.
-        results = dao.call_proc('sp_get_code_module_items', args=[self._module])
-        if results:
-            for row in results:
-                if row['code_type'] != 3:
-                    continue
+        # DEPRECATED after RDR 1.85.2:  Load module field data from the code table if available, using stored proc
+        # results = dao.call_proc('sp_get_code_module_items', args=[self._module])
+
+        # This query replaces the sp_get_code_module_items stored procedure, which does not support the
+        # DRC-managed codebooks where codes may be shared between modules. Columns are returned in the
+        # same order as the stored procedure returned them (as a debug aid for comparing results).
+        _question_codes_sql = """
+            select c.code_id,
+                   c.parent_id,
+                   c.topic,
+                   c.code_type,
+                   c.value,
+                   c.display,
+                   c.system,
+                   c.mapped,
+                   c.created,
+                   c.code_book_id,
+                   c.short_value
+            from code c
+            inner join (
+                select distinct qq.code_id
+                from questionnaire_question qq where qq.questionnaire_id in (
+                    select qc.questionnaire_id from questionnaire_concept qc
+                            where qc.code_id = (
+                                select code_id from code c2 where c2.value = :module_id and system = :system
+                            )
+                )
+            ) qq2 on qq2.code_id = c.code_id
+            where c.code_type = 3
```

Comment:
This will miss any codes that were previously Topic type codes, but have been repurposed as question codes. I think for the logic here it's safe to assume that the code type doesn't matter if the code_id is used by a questionnaire_question.

Reply:
Fixed in latest version